src/main/java/uk/gov/hmcts/befta/DefaultTestAutomationAdapter.java

```diff
@@ -105,16 +106,17 @@ protected void registerApiClientWithCredentials(String clientId, String clientSe
         tokenGenerators.put(clientId, tokenGenerator);
     }
 
-    private String getIdamOauth2Token(String username, String password) {
+    private String getIdamOauth2Token(String username, String password, OAuth2Config oauth2Config) {
         String authorisation = username + ":" + password;
         String base64Authorisation = Base64.getEncoder().encodeToString(authorisation.getBytes());
 
-        OAuth2 oauth2 = BeftaMain.getConfig().getOauth2Config();
         AuthApi.AuthenticateUserResponse authenticateUserResponse = idamApi
-                .authenticateUser(BASIC + base64Authorisation, CODE, oauth2.getClientId(), oauth2.getRedirectUri());
+                .authenticateUser(BASIC + base64Authorisation, CODE, oauth2Config.getClientId(),
+                        oauth2Config.getRedirectUri());
```

Comment:
Now there is one single api to get token rather than two calls.  (/o/token)

Reply:
RDM-8785 created for this refactoring.